lib/ts/recipe/totp/api/listDevices.ts

```diff
@@ -0,0 +1,38 @@
+/* Copyright (c) 2023, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { send200Response } from "../../../utils";
+import { APIInterface, APIOptions } from "../types";
+import { makeDefaultUserContextFromAPI } from "../../../utils";
+
+import Session from "../../session";
+
+export default async function listDevices(apiImplementation: APIInterface, options: APIOptions): Promise<boolean> {
+    if (apiImplementation.listDevicesGET === undefined) {
+        return false;
+    }
+
+    let session = await Session.getSession(options.req, options.res);
```

Comment:
Why? It should show device list only when someone is fully authenticated, right?

Reply:
Not necessarily. This API needs to be called from the frontend in order to know which flow to pick:
- Create device flow on the frontend
- Enter TOTP flow on the frontend.

Reply:
> which flow to pick

This will be handled by `/auth/mfa/factor/is-enabled` API which does this for TOTP:
```js
if (input.factorId === "totp") {
    const res = await listDevices({ userId: input.session.getUserId() });
    const verifiedDevicesCount = res.status === "OK" ? res.devices.filter((d) => d.verified).length : 0;
    return verifiedDevicesCount > 0;
}
```